openedx/features/caliper_tracking/transformers/enrollment_transformers.py

```diff
@@ -58,13 +58,28 @@ def edx_course_enrollment_mode_changed(current_event, caliper_event):
     :return: final created log
     """
     caliper_event.update({
-        'type': 'Enrollment',
-        'action': 'Mode_Changed',
-        'object': current_event['event'],
-        'course_id': current_event['context']['course_id']
+        'type': 'Event',
+        'action': 'Modified',
+        'object': {
+            'id': current_event['event']['course_id'],
+            'type': 'Membership'
+        },
+    })
+
+    caliper_event['actor'].update({
+        'type': 'Person',
+        'name': current_event['username']
+    })
+
+    caliper_event['referrer'].update({
+        'type': 'WebPage'
+    })
+
+    caliper_event['extensions']['extra_fields'].update({
+        'course_id': current_event['context']['course_id'],
+        'ip': current_event['ip']
```

Comment:
`'event_source': current_event['event_source']`

Add event_source for now in your extra fields